icgcget/cli.py

```diff
@@ -112,90 +113,93 @@ def cli(ctx, config, logfile):
 @click.option('--icgc-transport-file-from')
 @click.option('--icgc-transport-memory')
 @click.option('--icgc-transport-parallel')
-def download(repo, fileid, manifest, output,
+def download(repo, fileids, manifest, output,
              cghub_access, cghub_path, cghub_transport_parallel,
              ega_access, ega_password, ega_path, ega_transport_parallel, ega_udt, ega_username,
              gdc_access, gdc_path, gdc_transport_parallel, gdc_udt,
              icgc_access, icgc_path, icgc_transport_file_from, icgc_transport_memory, icgc_transport_parallel):
     logger = logging.getLogger('__log__')
-    code = 0
 
-    if fileid is None and manifest is None:
-        logger.error("Please provide either a file id value or a manifest file to download.")
-        code = 1
-        return code
-
-    if repo == 'ega':
+    object_ids = {}
+    if 'ega' in repo:
         if ega_username is None or ega_password is None:
-            if ega_access is None:
+            if ega_access is None:  # Do we want to raise exceptions on these, or just run them all?
                 logger.error("No credentials provided for the ega repository.")
-                code = 1
-                return code
-        if fileid is not None:
-            if len(fileid) > 1:
+                raise click.BadParameter
+    if 'collaboratory' in repo or 'aws-virginia' in repo:
+        if icgc_access is None:
```

Comment:
Again, helper method that can be reused at least 3 times here:

`check_credentials(icgc_access, 'icgc')`

Start thinking more opportunities where you can apply this pattern or something like it.